pkg/daemon/container-inspect.go

```diff
@@ -0,0 +1,77 @@
+package daemon
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/containers/image/v5/docker"
+	"github.com/containers/image/v5/image"
+	"github.com/containers/image/v5/types"
+	"github.com/pkg/errors"
+)
+
+func retryIfNecessary(ctx context.Context, operation func() error) error {
+	err := operation()
+	for attempt := 0; err != nil && attempt < numRetriesNetCommands; attempt++ {
+		delay := time.Duration(int(math.Pow(2, float64(attempt)))) * time.Second
+		fmt.Printf("Warning: failed, retrying in %s ... (%d/%d)", delay, attempt+1, numRetriesNetCommands)
+		select {
+		case <-time.After(delay):
+			break
+		case <-ctx.Done():
+			return err
+		}
+		err = operation()
+	}
+	return err
+}
+
+// parseImageSource converts image URL-like string to an ImageSource.
```

Comment:
With `docker.ParseReference` it is no longer very URL-like; and I would expect this function to handle the `"//"` prefix instead of callers having to care. Ideally, this should “convert an image reference”, or “create an `ImageSource` from an image reference”.